extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/photos/GooglePhotosInterface.java

```diff
@@ -180,12 +180,19 @@ BatchMediaItemResponse createPhoto(NewMediaItemUpload newMediaItemUpload) throws
     try {
       response = postRequest.execute();
     } catch (HttpResponseException e) {
-      // if the response is "unauthorized" and we've successfully refreshed the token, try the request again
-      if (e.getStatusCode() == 401 && credential.refreshToken()) {
-        // if the second attempt throws an error, then something else is wrong, and we bubble up the response errors
-        response = requestFactory
-            .buildPostRequest(new GenericUrl(url + "?" + generateParamsString(parameters)),
-                httpContent).execute();
+      // if the response is "unauthorized", refresh the token and try the request again
+      if (e.getStatusCode() == 401) {
+        monitor.info(() -> "Attempting to refresh authorization token");
+        if (credential.refreshToken()) {
```

Comment:
Do we need a pause here?

Reply:
I dont think so - refreshToken doesn't return until the token's been set if im reading this correctly: 
https://googleapis.dev/java/google-oauth-client/1.25.0/index.html (ctrl-f for "refreshtoken()")